### PR TITLE
Updated some gems: reek from 1.5.x to 1.6.x because it used an outdated

### PR DESCRIPTION
**Pull request by ralfclaassens**
Updated some gems: reek from 1.5.x to 1.6.x because it used an outdated

### CODE REVIEW

**Comment by joenas** [comment]
Hi,

The PR does more than just update reek version, any particular reason for that?
Also, it reintroduces `IrresponsibleModule`.

**Comment by ralfclaassens** [comment]
Hi Jon,

I think I did not reintroduced the IrresponsibleModule, but I've changed the rspecs because of the change of the underlying smell-class in Reek.

**Comment by joenas** [comment]
Yes, I see. I have in mind to remove the `-i` option in a major update to version `2.0.0` and just let users (mostly myself I believe) handle unwanted reports with `.reek` files instead. But I would prefer to keep the interface intact until then, so maybe you can change the code instead and leave the specs intact? It should be possible, I assume.

**Comment by ralfclaassens** [comment]
Thanks for the quick response! 

I'm sorry, that is a misunderstanding I think. The specs had to be changed, because they were failing due to the reek version-bump. In other words: it was not my intention to change them, but they had to....

**Comment by joenas** [comment]
Hi again,

Well what I'm trying to say is that it should be possible to change the code instead of the specs - unless they did something very inversive in `reek` -  because changing the specs means that the behaviour of the gems has changed, and I would prefer it not to. I am using the internals in `reek` though, which isn't optimal, but nonetheless. I'll have  a look at it later! :smile:

**Comment by ralfclaassens** [comment]
Hi Jon,

Ideally, I agree your opinion. However, it is simply not possible to keep the specs unchanged, because indeed, we're dependent on the external changes from reek.

File: lib/preek/examiner.rb [hunk 1]
```diff
@@ -32,7 +32,7 @@ def examine_and_report
 
     def filter_excludes_from(examiner)
       examiner.smells.delete_if do |smell|
-        @excludes.include? smell.smell_class
+        @excludes.include? smell.class
```
Comment (joenas) [inline lib/preek/examiner.rb]: If you change this to `@excludes.include? smell.smell_type`

File: lib/preek/ext/smell_warning.rb [hunk 2]
```diff
@@ -10,7 +10,7 @@ def smell_method
     end
 
     def smell_string
-      "#{smell_method} #{message} (#{subclass}) at lines #{Array(lines)*','}"
+      "#{smell_method} #{message} (#{smell_detector.class}) at lines #{Array(lines)*','}"
```
Comment (joenas) [inline lib/preek/ext/smell_warning.rb]: and this to `"#{smell_method} #{message} (#{smell_type}) at lines #{Array(lines)*','}"`
the specs should pass without changes.

File: lib/preek/examiner.rb (hunk 1 above)
Comment (ralfclaassens) [inline lib/preek/examiner.rb]: You're right. I've overseen that.

File: lib/preek/ext/smell_warning.rb (hunk 2 above)
Comment (ralfclaassens) [inline lib/preek/ext/smell_warning.rb]: You're right. I've overseen that. Corrected it.

File: spec/cli_spec.rb [hunk 3]
```diff
@@ -77,8 +77,8 @@ def test_file(file_name)
         Then{expect(output).to include(args[1])}
       end
 
-      context "when given file has Irresponsible smell" do
```
Comment (joenas) [inline spec/cli_spec.rb]: This change can be reverted as well, there is already a spec for non-smelly file [here](https://github.com/ralfclaassens/preek/blob/update_reek_version/spec/cli_spec.rb#L66).

Comment (ralfclaassens) [inline spec/cli_spec.rb]: You are right, I reverted this. Tests still pass by the way.

**Comment by joenas** [comment]
There you have it, nice work!

**Comment by joenas** [comment]
Thanks for the contribution :+1:

**Comment by ralfclaassens** [comment]
Great, how is releasing normally done?

**Comment by joenas** [comment]
> Great, how is releasing normally done?

Not really sure what you mean, but I usually release a new version when I make a (non-refactor) change, like now for example :smile:

**Comment by ralfclaassens** [comment]
Ok, thanks :+1:

**Pull request merged**